### DISCUSSION TITLE
Using event-driven functions with Symfony

### DISCUSSION THREAD

**Discussion by onokje** [brefphp/bref#1573] [Support] [answered]
I am trying to write an event-driven function, but I want to use Symfony. I tried the console layer, but it don't really understand it's 'magic'. I used https://bref.sh/docs/runtimes/function.html as a starting point, but I can't figure out how to use symfony this way. I want to be able to call the function manually and with a json payload and return value.
Can anyone point me in the right direction?

**Comment by mnapoli** [comment, marked as answer]
check this out: https://github.com/brefphp/symfony-bridge#class-handlers

**Comment by onokje** [comment]
This is exactly what I was looking for. Thanks!

**Reply by mnapoli** [reply]
Perfect! Are you building microservices with EventBridge or SQS?

**Comment by onokje** [comment]
For this particular case, neither, the function will be called manually on demand. But we are using both SQS and EventBridge as well.
